ArborX does not compile with C++20

`benchmarks` and `examples` are fine (at least using clang for Serial), but some errors in `tests/`, like
```
test/tstDetailsAlgorithms.cpp:30:23: error: no matching constructor for
initialization of 'Point' (aka 'Point<3>')
  BOOST_TEST(distance(Point{{1.0, 2.0, 3.0}}, Point{{1.0, 1.0, 1.0}}) ==
```

--- a/src/geometry/ArborX_HyperPoint.hpp
+++ b/src/geometry/ArborX_HyperPoint.hpp
@@ -22,9 +22,6 @@ namespace ArborX::ExperimentalHyperGeometry
 template <int DIM, class FloatingPoint = float>
 struct Point
 {
-  KOKKOS_DEFAULTED_FUNCTION
-  constexpr Point() noexcept = default;
-
   KOKKOS_FUNCTION
   constexpr auto &operator[](unsigned int i) { return _coords[i]; }
 

